### DISCUSSION TITLE
Markdown code block syntax highlighting issues in Quick Documentation

### DISCUSSION THREAD

**Discussion by harry-xm** [redhat-developer/lsp4ij#492] [Q&A]
`//` is not grayed out in the following parts of a `textDocument/hover` result, but is grayed out in the editor.

![Screenshot](https://github.com/user-attachments/assets/529c67e8-05c1-4a11-b415-f1a6c4b8017a)

```rust
// Implements notable traits: Future<Output = ?>
pub fn with_graceful_shutdown<F>(self, signal: F) -> Graceful<I, S, F, E>
where
    F: Future<Output = ()>,
    E: NewSvcExec<IO, S::Future, S::Service, E, GracefulWatcher>,
```

```rust
// Make a server from the previous examples...
let server = Server::bind(&([127, 0, 0, 1], 3000).into())
    .serve(make_service);

...
```

LSP server's response is

```
[Trace - 20:07:31] Received response 'textDocument/hover - (89)' in 1ms.
Result: {
  "contents": {
    "kind": "markdown",
    "value": "\n```rust\nhyper::server::server::Server\n```\n\n```rust\n// Implements notable traits: Future\u003cOutput \u003d ?\u003e\npub fn with_graceful_shutdown\u003cF\u003e(self, signal: F) -\u003e Graceful\u003cI, S, F, E\u003e\nwhere\n    F: Future\u003cOutput \u003d ()\u003e,\n    E: NewSvcExec\u003cIO, S::Future, S::Service, E, GracefulWatcher\u003e,\n```\n\n---\n\nPrepares a server to handle graceful shutdown when the provided future\ncompletes.\n\n# Example\n\n```rust\n// Make a server from the previous examples...\nlet server \u003d Server::bind(\u0026([127, 0, 0, 1], 3000).into())\n    .serve(make_service);\n\n// Prepare some signal for when the server should start shutting down...\nlet (tx, rx) \u003d tokio::sync::oneshot::channel::\u003c()\u003e();\nlet graceful \u003d server\n    .with_graceful_shutdown(async {\n        rx.await.ok();\n    });\n\n// Await the `server` receiving the signal...\nif let Err(e) \u003d graceful.await {\n    eprintln!(\"server error: {}\", e);\n}\n\n// And later, trigger the signal by calling `tx.send(())`.\nlet _ \u003d tx.send(());\n```"
  },
  "range": {
    "start": {
      "line": 96,
      "character": 9
    },
    "end": {
      "line": 96,
      "character": 31
    }
  }
}
```

**Comment by angelozerr** [comment]
could you share your file please?

**Reply by harry-xm** [reply]
Sure, here is the link: https://github.com/eunomia-bpf/eunomia-bpf/blob/291304bf415817d2f5b46a3655d9f9594b8672a6/ecli/server/src/main.rs#L97

**Reply by harry-xm** [reply]
After disabling the bundled plugin "TextMate Bundles", the syntax highlighting is gone.

Therefore, the actual question is whether syntax highlighting of code blocks in hover popup should be handled by LSP4IJ or not.

**Reply by angelozerr** [reply]
LSP4IJ can manage syntax highlighting by using standard IJ syntax component (ex: TextMate, custom IJ syntax coloration by using the PsiFile), but it cannot manage LSP semantic tokens highligting in code block. I don't know how to manage that, any contribution are welcome!

**Comment by angelozerr** [comment]
I have tested and it works for me:

![image](https://github.com/user-attachments/assets/7ff637f5-377f-4219-a0d2-79b2d1f03f87)

I wonder if you are using an another IJ plugin which does some syntax coloration (and not textmate) ?

**Reply by harry-xm** [reply]
I don't have other non-bundled plugins installed. Maybe it's related to the IDE or OS? I'm running PyCharm (Community Edition) on macOS.

PyCharm 2024.2.0.1 (Community Edition)
Build #PC-242.20224.428, built on August 21, 2024
Runtime version: 21.0.3+13-b509.4 aarch64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.lwawt.macosx.LWCToolkit
macOS 14.6.1
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 2048M
Cores: 8
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
Non-Bundled Plugins:
  com.redhat.devtools.lsp4ij (0.4.0)

**Comment by angelozerr** [comment]
I close this discussion in favor of https://github.com/redhat-developer/lsp4ij/issues/557
